arrow_chunk.go

```diff
@@ -94,9 +94,10 @@ func (arc *arrowResultChunk) passRawArrowBatch(scd *snowflakeChunkDownloader) (*
 		}
 		// Here we check all metadata from snowflake are preserved, so evaluator can decode accordingly
 		for idx, field := range rawRecord.Schema().Fields() {
-			if field.Nullable != scd.RowSet.RowType[idx].Nullable ||
-				field.Name != scd.RowSet.RowType[idx].Name ||
```

Comment:
Do you know why the check `field.Name != scd.RowSet.RowType[idx].Name` is here to begin with? I'd be careful about removing things that seem useless without first understanding why they might be useful.

Reply:
This was my initial decision to compare all metadata inside the driver before sending the records over. And now I feel like it is not really needed.
I found this bug when I ran this `select to_varchar(...);` query. Snowflake gives us arrow records specified as nullable, while the `rowtype` metadata has nullable as false. This query should work fine if we stop treating this mismatch as a fatal error - we should be able to solely depend on the arrow records attribute, and even if Snowflake has an internal error and thus sends us with non-nullable records containing nulls, this will be detected in evaluator and bubble up.

Reply:
This is causing problems because now such queries will always error out though it should be able to work.